modules/genomic_browser/php/NDB_Menu_Filter_cpg_browser.class.inc

```diff
@@ -0,0 +1,429 @@
+     * @return bool
+     */
+    function _setFilterForm()
+    {
+        // create user object
+        $user = User::singleton();
+
+        // PSC
+        if ($user->hasPermission('genomic_browser_view_allsites')) {
+            // has access to all sites genomic data
+            // get the list of study sites - to be replaced by the Site object
+            $list_of_sites = Utility::getSiteList();
+            if (is_array($list_of_sites)) {
+                $list_of_sites = array('' => 'Any') + $list_of_sites;
+            }
+        } else {
+            // allow only to view own site data
+            $site = Site::singleton($user->getData('CenterID'));
+            if ($site->isStudySite()) {
+                $list_of_sites
+                    = array($user->getData('CenterID') => $user->getData('Site'));
+            }
+        }
+
+        // SubprojectID
+        $list_of_subprojects = Utility::getSubprojectList();
+
+        // add form elements
+        $this->addSelect('centerID', 'Site:', $list_of_sites);
+        $this->addBasicText('DCCID', 'DCCID:');
+        $this->addBasicText('PSCID', 'PSCID:');
+        $this->addBasicText('External_ID', 'External ID:');
+        $this->addSelect(
+            'gender',
+            'Gender:',
+            array(
+             ''       => 'All',
+             'Male'   => 'Male',
+             'Female' => 'Female',
+            )
+        );
+        $this->addSelect(
+            'SubprojectID',
+            'Subproject:',
+            array('' => 'Any') + $list_of_subprojects
+        );
+        $this->addBasicText('dob', 'Date of Birth:', array('maxlength' => 10));
+
+        $show_results_options
+            = array(
+               'brief' => 'Summary fields',
+               'full'  => 'All fields',
+              );
+        $this->addSelect('Show_Brief_Results', 'Display:', $show_results_options);
+
+        // CpG
+        $this->addBasicText('CpG', 'CpG:');
+        $Base_options = array(
+                         null      => 'Any',
+                         'Island'  => 'Island',
+                         'N_Shore' => 'North shore',
+                         'S_Shelf' => 'South shelf',
+                         'S_Shore' => 'South shore',
+                        );
+        $this->addSelect('Context', 'Context:', $Base_options);
+        $this->addBasicText('Date_collected', 'Date collected:');
+        $Base_options = array(
+                         null               => 'Any',
+                         'High-methylation' => 'High-methylation',
+                         'Partially'        => 'Partially',
+                         'Low-methylation'  => 'Low-methylation',
+                        );
+        $this->addSelect('Methylation', 'Methylation:', $Base_options);
+        $this->addBasicText('Beta_value', 'Beta-value:');
+        $Base_options = array(
+                         null     => 'Any',
+                         'Buccal' => 'Buccal',
```

Comment:
I agree that this should come from the database.
I can be all, none or some of them. This will depend on the study uploading file format.